### ISSUE TITLE
Request for plugin support: targets.vim

### ISSUE THREAD

**Issue by extr0py** [VSCodeVim/Vim#601]
One plugin that is very useful for my VIM flow is this one:
https://github.com/wellle/targets.vim

It builds on the idea of the ```ci"``` command - where it jumps forward to find the nearest quotation, and changes inside, for other characters, like (), {}, [], etc.

It also supports argument text objects - which are handy for deleting and changing in function arguments. So ```cia``` would jump forward to an argument and change. 

These motions and text objects are nice because they mean you can strike from a distance.

In terms of integration, it seems like it depends on whether the plugin story for this is implemented:
https://github.com/VSCodeVim/Vim/issues/590

If that happens, we might want to look at making the text objects extensible to support these scenarios.
Otherwise, we could potentially bake it in as a setting.

Note that ```ci"``` is already supported in VSCodeVim (and VIM out of the box), so potentially the work at least for those jump commands might not be too bad.

**Comment by johnfn** [comment]
Anyone who wants this feature can feel free to get started on it. It would simply be too much work for me to port over every Vim plugin, but if you (or anyone else) wants to open up a WIP PR, or ask me questions on Slack, I'm happy to help you along the way. :)

**Comment by Austin-Scott** [comment]
Inspired by the recent addition of argument text objects #4653 I'm currently taking a shot at implementing the `in` ("inner next" such as `cin(`), `an` ("around next" such as `can(`), `il` ("inner last" such as `cil(`), and `al` ("around last" such as `cal(`) commands from this plugin.

**Comment by J-Fields** [comment]
@Austin-Scott Sweet! Feel free to reach out if you need any guidance

**Comment by berknam** [comment]
> 
> 
> Inspired by the recent addition of argument text objects #4653 I'm currently taking a shot at implementing the `in` ("inner next" such as `cin(`), `an` ("around next" such as `can(`), `il` ("inner last" such as `cil(`), and `al` ("around last" such as `cal(`) commands from this plugin.

I've been looking into implementing the `:normal` command which would allow to create this type of text objects with operator-pending mode remaps (after PR #4735) which would allow to create remaps like `onoremap in( :<C-u>normal! f(vi(<CR>`  and `onoremap il( :<C-u>normal! F)vi(<CR>` 

It might take a while still.

**Comment by larrywal** [comment]
Any news or progress here? - agree this would be super useful.  In particular, the seeking behavior where I don't have to be within brackets or braces to make something like ci( work.

**Comment by Austin-Scott** [comment]
Sadly no. I haven't made any progress in the last few months. I still hope to eventual finish this, but I can't give any predictions as to when that will happen right now.

**Comment by elazarcoh** [comment]
I just made a PR #7025 for the quotes text object of **targets.vim**.
Will gladly take a shot at implementing the `cin(` etc. as well, if @Austin-Scott feels it's not likely to happen any time soon.

**Comment by elazarcoh** [comment]
@J-Fields (you seems like the main maintainer here, so I'm tagging you..)

Hi,
First, I just want you to know that I really appreciate all the work you and other collaborators put into VSCodeVim.
You probably have a lot on your plate, I don't want to sound like I'm rushing or something.
I just want to ask, since contributions are expected to be reviewed, when I can expect someone to get starting on it? 
Maybe you can have some insiders version or something, so you can get PR faster into the insiders version?


I already implemented the `din(` etc., can be found here: https://github.com/elazarcoh/Vim/tree/last-next-objects (I don't want to make another PR for it just yet, only after we start on the first one).

Please let us know if it can get scheduled any time soon.

Thanks

**Comment by J-Fields** [comment]
@elazarcoh Sorry for the delay - I'm notoriously slow at getting to PRs and have been busy with other things lately. No promises, but I'll do my best to get at least a first pass review done this weekend.

> Maybe you can have some insiders version or something, so you can get PR faster into the insiders version?

No insiders version yet - I've been waiting on first-class support from the VSCode team which, coincidentally, seems to be making some progress (https://github.com/microsoft/vscode/issues/15756#issuecomment-916728348). We can cut a new version once the PR is merged, though.

**Comment by numToStr** [comment]
First class vim support sounds super cool 😍.

**Comment by elazarcoh** [comment]
The most useful text-objects from targets.vim (at least for me) are implemented:
* quotes #7025
* last/next brackets #7558
* arguments (not as a plugin)

there are a few more that are not implemented yet:
* 👀 support for count in the above objects
* 🎉 separator text object (see targets.vim github for explanation)
* 😄 general bracket object: `b` for the first match of ([{
* 🚀 last/next html tag
* ❤️ last/next argument

I don't know if I'll do all of the above, but I wanted to know which one will be the most useful to you, so I'll do it next. please upvote by the emoji.
